### PR TITLE
Rename Complex.power as Complex.pow

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
See https://github.com/elixir-nx/nx/issues/982.

### CODE REVIEW

**Comment by josevalim** [comment]
For backwards compatibility, can we please keep `Complex.power` around which simply delegates to `pow`? We can add a deprecation warning. That's how we will do things in Nx too.

**Comment by Benjamin-Philip** [comment]
Yeah sure.

File: lib/complex.ex [hunk 1]
```diff
@@ -1042,12 +1042,15 @@ defmodule Complex do
...
-        s = multiply(power(rho, y.re), exp(multiply(negate(y.im), theta)))
+        s = multiply(pow(rho, y.re), exp(multiply(negate(y.im), theta)))
         r = add(multiply(y.re, theta), multiply(y.im, ln(rho)))
         new(multiply(s, cos(r)), multiply(s, sin(r)))
     end
   end
 
+  @deprecated "Use pow/2 instead"
+  def power(x, y), do: pow(x, y)
+
```
Comment (Benjamin-Philip) [inline lib/complex.ex:1053]: @josevalim, Does this message look good? I directly copy-pasted it from https://elixir-lang.org/blog/2018/01/17/elixir-v1-6-0-released/

**Comment by Benjamin-Philip** [comment]
If you release a new minor version, I can update the instances in Nx as well in my Nx.pow PR.

File: lib/complex.ex (hunk 1 above)
Comment (josevalim) [reply lib/complex.ex:1053]: Yup, thanks!

**Review by josevalim — APPROVED** [review]

**Comment by Benjamin-Philip** [comment]
Hey @polvalente, if you merge and release a new minor release, I can update `Complex` before `Nx`'s v0.5 release.

**Review by polvalente — APPROVED** [review]
If we're going down this route we should also rename ln to log, although I really hate this convention. Do you want to submit a PR as well?

**Comment by Benjamin-Philip** [comment]
> If we're going down this route we should also rename ln to log, although I really hate this convention. Do you want to submit a PR as well?

Personally, I don't like the `pow` convention at all. It's just that it's followed in Elixir core, so we have to follow it.

**Comment by polvalente** [comment]
> > If we're going down this route we should also rename ln to log, although I really hate this convention. Do you want to submit a PR as well?
> 
> Personally, I don't like the `pow` convention at all. It's just that it's followed in Elixir core, so we have to follow it. 

Likewise, Nx.log is the function for ln because of :math.log, so it would make sense to rename Complex.ln

**Comment by Benjamin-Philip** [comment]
> Likewise, Nx.log is the function for ln because of :math.log, so it would make sense to rename Complex.ln

Yeah, I understood that the first time, I'm working on it.
Expect a PR by tomorrow.

**Pull request merged**